federator/src/lib/FederatorERC.ts

```diff
@@ -0,0 +1,614 @@
+        );
+        await this._voteTransaction({
+          sideChainId,
+          mainChainId,
+          transactionSender,
+          sideFedContract,
+          tokenAddress,
+          sender: crossFrom,
+          receiver,
+          amount,
+          symbol,
+          blockHash: log.blockHash,
+          transactionHash: log.transactionHash,
+          logIndex: log.logIndex,
+          decimals,
+          granularity,
+          typeId,
+          txId: transactionId,
+          federatorAddress,
+          originChainId,
+          destinationChainId,
+          sideChainConfig,
+        });
+      } else {
+        this.logger.debug(
+          `Block: ${log.blockHash} Tx: ${log.transactionHash} originalTokenAddress: ${tokenAddress}  has already been voted by us`,
+        );
+      }
+    } else {
+      this.logger.debug(
+        `Block: ${log.blockHash} Tx: ${log.transactionHash} originalTokenAddress: ${tokenAddress} was already processed`,
+      );
+    }
+  }
+
+  async _processLogs({
+    sideChainId,
+    mainChainId,
+    transactionSender,
+    logs,
+    currentBlock,
+    mediumAndSmall,
+    confirmations,
+    sideChainConfig,
+    federationFactory,
+    allowTokensFactory,
+    bridgeFactory,
+  }: {
+    sideChainId: number;
+    mainChainId: number;
+    transactionSender: TransactionSender;
+    logs: any;
+    currentBlock: number;
+    mediumAndSmall: boolean;
+    confirmations: { mediumAmountConfirmations: number; largeAmountConfirmations: number };
+    sideChainConfig: ConfigChain;
+    federationFactory: FederationFactory;
+    allowTokensFactory: AllowTokensFactory;
+    bridgeFactory: BridgeFactory;
+  }) {
+    try {
+      const federatorAddress = await transactionSender.getAddress(this.config.privateKey);
+      const sideFedContract = await federationFactory.getSideFederationContract();
+      const allowTokens = await allowTokensFactory.getMainAllowTokensContract();
+
+      await this.checkFederatorIsMember(sideFedContract, federatorAddress);
+
+      for (const log of logs) {
+        await this.processLog({
+          sideChainId,
+          mainChainId,
+          transactionSender,
+          log,
+          confirmations,
+          sideFedContract,
+          allowTokens,
+          currentBlock,
+          federatorAddress,
+          mediumAndSmall,
+          sideChainConfig,
+          allowTokensFactory,
+          bridgeFactory,
+        });
+      }
+
+      return true;
+    } catch (err) {
+      throw new CustomError(`Exception processing logs`, err);
+    }
+  }
+
+  async _voteTransaction({
+    sideChainId,
+    mainChainId,
+    transactionSender,
+    sideFedContract,
+    tokenAddress,
+    sender,
+    receiver,
+    amount,
+    symbol,
+    blockHash,
+    transactionHash,
+    logIndex,
+    decimals,
+    granularity,
+    typeId,
+    txId,
+    federatorAddress,
+    originChainId,
+    destinationChainId,
+    sideChainConfig,
+  }: {
+    sideChainId: number;
```

Comment:
Maybe we should create the object with this properties separated from the function to avoid having so much stuff.
Also this type definition could be shared with _processLogs.
I know they have different properties, maybe creating one for processLogs and then adding to that the necesary for _voteTransaction it's better

Reply:
done at https://github.com/rsksmart/tokenbridge/pull/289/commits/eb40bd7d19842c74ab2bf0ac7565566e58d5e406